app/src/main/java/com/example/dogbook/SignupActivity/SignupActivity.java

```diff
@@ -3,15 +3,122 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 
+import android.content.Intent;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.Toast;
 
+import com.example.dogbook.LoginActivity.LoginActivity;
+import com.example.dogbook.MainActivity.MainActivity;
 import com.example.dogbook.R;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+import java.util.Date;
 
 public class SignupActivity extends AppCompatActivity {
 
+    private static final String TAG = "SignupActivity";
+
+    EditText etUsername;
+    EditText etPassword;
+    EditText etConfirmPassword;
+    EditText etBreed;
+    DatePicker dpBirthday;
+    CheckBox cbLookingForPlaymates;
+    EditText etOwnerName;
+    EditText etOwnerContact;
+    Button btnSignup;
+    Button btnLogin;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_signup);
+        initView();
+        btnSignup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signUp();
+            }
+        });
+
+        btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goLoginActivity();
+            }
+        });
+
+    }
+
+    private void initView() {
+        etUsername = findViewById(R.id.etUsername);
+        etPassword = findViewById(R.id.etPassword);
+        etConfirmPassword = findViewById(R.id.etConfirmPassword);
+        etBreed = findViewById(R.id.etBreed);
+        dpBirthday = findViewById(R.id.dpBirthday);
+        cbLookingForPlaymates = findViewById(R.id.cbLookingForPlaymates);
+        etOwnerName = findViewById(R.id.etOwnerName);
+        etOwnerContact = findViewById(R.id.etOwnerContact);
+        btnSignup = findViewById(R.id.btnSignup);
+        btnLogin = findViewById(R.id.btnLogin);
+    }
+
+    private void signUp() {
+        //Password confirmation was invalid
+        if (!etPassword.getText().toString().equals(etConfirmPassword.getText().toString())) {
```

Comment:
Is it possible that either of the `getText()` methods can return `null`?  If so, you would end up with a null pointer exception here

Reply:
I don't think so, I tried it and apparently it would return an empty string